geode-core/src/main/java/org/apache/geode/internal/cache/DistributedRegion.java

```diff
@@ -526,21 +526,36 @@ public boolean hasSeenEvent(EntryEventImpl event) {
     isDuplicate = getEventTracker().hasSeenEvent(event);
     if (isDuplicate) {
       markEventAsDuplicate(event);
+      if (event.getVersionTag() == null && event.getRegion().getConcurrencyChecksEnabled()
+          && event.getEventId() != null) {
+        // For a retry, versionTag can be missing possibly due to no versionTag recorded during gii.
+        findAndSetVersionTag(event);
+        if (event.getVersionTag() == null) {
+          logger.info("No version tag can be found in cluster when retrying the following event {}",
+              event);
```

Comment:
Is this likely to be logged often? If there are situations where it could be logged a lot, it might be better to make this debug level.

Reply:
No, we should not see this -- only a very corner case.